x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -358,31 +369,50 @@ export const entryHasListType = (
  * Returns the value for `file.Ext.code_signature` which
  * can be an object or array of objects
  */
-export const getCodeSignatureValue = (
-  alertData: Ecs
+export const getFileCodeSignature = (
+  alertData: FlattenType<Ecs>
 ): Array<{ subjectName: string; trusted: string }> => {
   const { file } = alertData;
   const codeSignature = file && file.Ext && file.Ext.code_signature;
 
-  // Pre 7.10 file.Ext.code_signature was mistakenly populated as
-  // a single object with subject_name and trusted.
+  return getCodeSignatureValue(codeSignature);
+};
+
+/**
+ * Returns the value for `process.Ext.code_signature` which
+ * can be an object or array of objects
+ */
+export const getProcessCodeSignature = (
+  alertData: FlattenType<Ecs>
+): Array<{ subjectName: string; trusted: string }> => {
+  const { process } = alertData;
+  const codeSignature = process && process.Ext && process.Ext.code_signature;
+  return getCodeSignatureValue(codeSignature);
+};
+
+/**
+ * Pre 7.10 `Ext.code_signature` fields were mistakenly populated as
+ * a single object with subject_name and trusted.
+ */
+export const getCodeSignatureValue = (
+  codeSignature: FlattenType<CodeSignature> | FlattenType<CodeSignature[]> | undefined
+): Array<{ subjectName: string; trusted: string }> => {
   if (Array.isArray(codeSignature) && codeSignature.length > 0) {
-    return codeSignature.map((signature) => ({
-      subjectName: (signature.subject_name && signature.subject_name[0]) ?? '',
-      trusted: (signature.trusted && signature.trusted[0]) ?? '',
-    }));
+    return codeSignature.map((signature) => {
+      return {
+        subjectName: signature.subject_name ?? '',
+        trusted: signature.trusted ?? '',
```

Comment:
Returning a empty string when a string is expected seems a bit like code smell. Would code that uses `getCodeSignatureValue` expect and handle empty strings? What special meaning does an empty string have?